src/dnaio/multipleend.py

```diff
@@ -0,0 +1,205 @@
+                f"an unequal amount of reads.",
+                line=None,
+            )
+
+    def close(self):
+        for reader in self.readers:
+            reader.close()
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *exc):
+        self.close()
+
+
+class MultipleFastaWriter(MultipleFileWriter):
+    def __init__(
+        self,
+        *files: Union[str, PathLike, BinaryIO],
+        opener=xopen,
+        append: bool = False,
+    ):
+        if len(files) < 1:
+            raise ValueError("At least one file is required")
+        mode = "a" if append else "w"
+        self.files = files
+        self.number_of_files = len(files)
+        self.writers: List[Union[FastaWriter, FastqWriter]] = [
+            _open_single(  # type: ignore
+                file,
+                opener=opener,
+                fileformat="fasta",
+                mode=mode,
+                qualities=False,
+            )
+            for file in self.files
+        ]
+
+    def __repr__(self) -> str:
+        return (
+            f"{self.__class__.__name__}"
+            f"({', '.join(repr(writer) for writer in self.writers)})"
+        )
+
+    def close(self):
+        for writer in self.writers:
+            writer.close()
+
+    def write(self, *records: SequenceRecord):
+        if len(records) != self.number_of_files:
+            raise ValueError(f"records must have length {self.number_of_files}")
+        for record, writer in zip(records, self.writers):
+            writer.write(record)
+
+    def write_iterable(self, records_iterable: Iterable[Tuple[SequenceRecord, ...]]):
+        for records in records_iterable:
+            self.write(*records)
+
+    def __enter__(self):
+        return self
+
+    def __exit__(self, *exc):
+        self.close()
+
+
+class MultipleFastqWriter(MultipleFileWriter):
+    def __init__(
+        self,
+        *files: Union[str, PathLike, BinaryIO],
+        opener=xopen,
+        append: bool = False,
+    ):
+        if len(files) < 1:
+            raise ValueError("At least one file is required")
+        mode = "a" if append else "w"
+        self.files = files
+        self.number_of_files = len(files)
+        self.writers: List[IO] = [
+            opener(file, mode + "b") if not hasattr(file, "write") else file
+            for file in self.files
+        ]
+
+    def __repr__(self) -> str:
+        return f"{self.__class__.__name__}" f"({', '.join(str(f) for f in self.files)})"
+
+    def close(self):
+        for writer in self.writers:
+            writer.close()
+
+    def write(self, *records: SequenceRecord):
+        if len(records) != self.number_of_files:
+            raise ValueError(f"records must have length {self.number_of_files}")
+        for record, writer in zip(records, self.writers):
+            writer.write(record.fastq_bytes())
+
+    def write_iterable(self, records_iterable: Iterable[Tuple[SequenceRecord, ...]]):
```

Comment:
This needs a docstring that describes when to use this method over `write()`.

I’m not so fond of the name; would something like `write_many` be clearer? (Not perfect either, though.)

Reply:
write_multiple? 
write_list?
Write list sounds quite okay, and properly conveys that the list can also have one entry. That it accepts any iterable container type should not really be that important.

Reply:
Hm, I think `write_iterable` is still better than those alternatives. Perhaps the method just needs a docstring and then it’s fine.

Reply:
The method has a docstring trough the MultipleFileWriter interface. It is an abstractmethod that is inherited.